_posts/2023-05-17-otp-26-highlights.md

```diff
@@ -0,0 +1,892 @@
+```
+
+Since the command line was valid, `argparse` collected the arguments
+into a map, which was then printed by the `handler` fun.
+
+The number of lines to be printed from each file defaults to `10`, but
+can be changed using either the `-n` or `--lines` option:
+
+```
+$ ehead -n 42 foo bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead foo --lines=42 bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead --lines 42 foo bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead foo bar --lines 42 baz
+#{lines => 42,files => ["foo","bar","baz"]}
+```
+
+Attempting to give the number of lines as `0` results in an error message:
+
+```
+$ ehead -n 0 foobar
+error: ehead: invalid argument for lines: 0 is less than accepted minimum
+Usage:
+  ehead [-n <lines>] [--lines <lines>] <files>...
+
+Arguments:
+  files       files
+
+Optional arguments:
+  -n, --lines number of lines to print (int >= 1, 10)
+```
+
+## Pull request
+
+[[argparse] Command line parser for Erlang](https://github.com/erlang/otp/pull/6852)
+
+
+# SSL: Safer defaults
+
+In OTP 25, the default client options for the `ssl` appliction would
+allow setting up a connection without verifying that the server used
+trusted CA certificates. For example:
+
+```erlang
+Erlang/OTP 25 . . .
+
+Eshell V13.1.1  (abort with ^G)
+1> application:ensure_all_started(ssl).
+{ok,[crypto,asn1,public_key,ssl]}
+2> ssl:connect("www.erlang.org", 443, []).
+=WARNING REPORT==== 6-Apr-2023::12:29:20.824457 ===
+Description: "Authenticity is not established by certificate path validation"
+     Reason: "Option {verify, verify_peer} and cacertfile/cacerts is missing"
+
+{ok,{sslsocket,{gen_tcp,#Port<0.6>,tls_connection,undefined},
+               [<0.122.0>,<0.121.0>]}}
+```
+
+A warning report would be generated, but a connection would be set up.
+
+In OTP 26, the default value for the `verify` option is now
+`verify_peer` instead of `verify_none`. Host verification
+requires the `cacert` option to also be given, which means that
+trying to set up a connection with an empty option list fails in
+OTP 26:
+
+```erlang
+Erlang/OTP 26 . . .
+
+Eshell V14.0 (press Ctrl+G to abort, type help(). for help)
+1> application:ensure_all_started(ssl).
+{ok,[crypto,asn1,public_key,ssl]}
+2> ssl:connect("www.erlang.org", 443, []).
+{error,{options,incompatible,
+                [{verify,verify_peer},{cacerts,undefined}]}}
+```
+
+The default value for the `cacerts` option is `undefined`,
+which is not compatible with the `{verify,verify_peer}` option.
+
+To make the connection succeed, the recommended way is to
+use the `cacerts` option to supply CA certificates to be used
+for verifying. For example:
+
+```erlang
+1> application:ensure_all_started(ssl).
+{ok,[crypto,asn1,public_key,ssl]}
+2> ssl:connect("www.erlang.org", 443, [{cacerts, public_key:cacerts_get()}]).
+{ok,{sslsocket,{gen_tcp,#Port<0.5>,tls_connection,undefined},
+               [<0.137.0>,<0.136.0>]}}
+```
+
```

Comment:
Also a lot of leagacy algorithms have been removed from default values such as sha1 and dsa to mention a few!

Reply:
I have added a sentence about that.